src/views/preview/preview.jsx

```diff
@@ -463,26 +451,83 @@ const consolidateStudiosInfo = (curatedStudios, projectStudios, currentStudioIds
     return consolidatedStudios;
 };
 
-const mapStateToProps = state => ({
-    comments: state.preview.comments,
-    faved: state.preview.faved,
-    loved: state.preview.loved,
-    original: state.preview.original,
-    parent: state.preview.parent,
-    remixes: state.preview.remixes,
-    replies: state.preview.replies,
-    sessionStatus: state.session.status,
-    projectInfo: state.preview.projectInfo,
-    projectStudios: state.preview.projectStudios,
-    studios: consolidateStudiosInfo(state.preview.curatedStudios,
-        state.preview.projectStudios, state.preview.currentStudioIds,
-        state.preview.status.studioRequests),
-    user: state.session.session.user,
-    playerMode: state.scratchGui.mode.isPlayerOnly,
-    fullScreen: state.scratchGui.mode.isFullScreen
-});
+const mapStateToProps = state => {
+    const projectInfoPresent = Object.keys(state.preview.projectInfo).length > 0;
+    const userPresent = state.session.session.user &&
+        Object.keys(state.session.session.user).length > 0;
+    const isLoggedIn = state.session.status === sessionActions.Status.FETCHED &&
+        userPresent;
+    const authorPresent = projectInfoPresent && state.preview.projectInfo.author &&
+        Object.keys(state.preview.projectInfo.author).length > 0;
+
+    return {
+        comments: state.preview.comments,
+        faved: state.preview.faved,
+        fullScreen: state.scratchGui.mode.isFullScreen,
+        // TODO: this value's logic is unfinished. Currently it seems to be
+        // identical to userOwnsProject, but we need to also set this true
+        // for admins and mods.
```

Comment:
We do have this information in session! In `if (permissions.admin === true)`

Reply:
Changed the logic to use that, take a look please.